apps/backoffice-v2/src/lib/react-query/query-client.ts

```diff
@@ -27,28 +33,34 @@ export const queryClient = new QueryClient({
     },
   },
   queryCache: new QueryCache({
-    onError: async error => {
-      if (isErrorWithCode(error) && error.code === 401) {
-        const authenticatedUser = authQueryKeys.authenticatedUser();
-
-        void queryClient.cancelQueries();
-        queryClient.setQueryData(authenticatedUser.queryKey, {
-          user: undefined,
-        });
-      }
-
+    onError: (error: unknown) => {
       if (isZodError(error)) {
         toast.error(t('toast:validation_error'));
+      }
 
-        return;
+      if (!isErrorWithCode(error)) {
+        throw error;
       }
 
-      if (!isErrorWithMessage(error) || error.message === 'undefined' || error.message === 'null')
-        return;
+      if (isErrorWithCode(error)) {
+        const status = error.code;
+        if (status === 401) {
+          void clearAuthenticatedUser(queryClient);
+        }
 
-      toast.error(error.message, {
-        id: error.message,
-      });
+        if (
+          isErrorWithMessage(error) &&
+          error.message !== 'undefined' &&
+          error.message !== 'null'
+        ) {
+          // Dont toast for no important errors
+          if (![401, 403, 404].includes(status)) {
+            toast.error(error.message, {
+              id: error.message,
+            });
+          }
+        }
+      }
```

Comment:
This block is getting hard to read. My suggestion is the extreme to the other direction so feel free to say if you see a middle-ground.
```suggestion
if (!isErrorWithCode(error)) return;

const status = error.code;

if (status === 401) {
  void clearAuthenticatedUser(queryClient);
}

if (!isErrorWithMessage(error) || error.message === 'undefined' || error.message === 'null') {
  return;
}

// Dont toast for no important errors
if ([401, 403, 404].includes(status)) return;

toast.error(error.message, {
  id: error.message,
});
```